Adiciona post sobre contribuição para Open Source com Github (#5)
OpenDevUFCG/OpenDevUFCGBlog: ContribuicaoOpenSource.md

---

> title: Contribuindo para projetos Open Source com Github

Reviewer: Estilizar para "GitHub" onde tiver o nome da plataforma e acho que seria melhor deixar open source em minúsculo ao longo do texto também.

published: false

> description: Uma porta de entrada para pessoas que planejam começar a fazer contribuições para organizações Open Source, falando um pouco sobre o que é o Github (uma visão geral sobre o que é a plataforma, sem um aprofundamento muito grande em detalhes técnicos) e como possibilita o crescimento de comunidades open source. Além disso, este post mostra também como ficar atento ao surgimento de novas oportunidades de contribuição em projetos.

Reviewer: É interessante que a descrição seja curta e objetiva (algo não muito maior que o título). Essa descrição é a que é exibida, por exemplo, quando se publica no Twitter. Um exemplo de bom tamanho pra descrição:
![image](https://user-images.githubusercontent.com/20714148/62173940-5c3e1f80-b30e-11e9-8269-f7f705132474.png)

> tags: opensource, comunidade, github.

Reviewer: Lembra de colocar a tag **ptbr**. Como o dev.to é um site predominantemente em inglês, é bom que as tags estejam em inglês mesmo (então seria bom mudar comunidade para community.

cover_image: https://thepracticaldev.s3.amazonaws.com/i/16tq00lwa6aq3wak8jsf.jpg
---

**Atenção: este post não tem a intenção de se aprofundar a respeito do funcionamento do Git e do Github, caso deseje informações mais técnicas, [acesse este outro post](https://medium.com/@Juliobguedes/entendendo-git-883464f379de).**

> Criado em 2008 e muito aperfeiçoado desde então, o Github não apenas é, atualmente, o mais conhecido serviço web que oferece diversas funcionalidades aplicadas ao Git para hospedagem e versionamento de código-fonte, como também é um grande impulsionador da iniciativa Open Source e do trabalho em equipe. Cada vez mais, a plataforma tem adquirido uma cara de rede social, visando engajar mais pessoas em relação a contribuição em projetos de código aberto, participação em suas respectivas comunidades, além de dar mais visibilidade a tais projetos. Isso foi possível devido a inúmeras funcionalidades adotadas e implementadas pelo Github a fim de atingir esses objetivos, e algumas delas serão introduzidas nesse post.

Reviewer 2: Corrige pra "... o GitHub não apenas é, ..., como também um grande ...", lembra sempre, GitHub com G e H maiúsculos

*Screenshot da página inicial do Github atualmente*.

## Stars e forks
![Stars e forks](https://thepracticaldev.s3.amazonaws.com/i/pu4yui075180v6w6k318.png)

Ao acessar um repositório público no Github, você pode dar uma **star**, *feature* similar às "curtidas" das redes sociais mais famosas, e também pode fazer um **fork**, que consiste na criação de uma cópia independente deste repositório, que contém todo o seu conteúdo, para a sua conta de usuário no Github. Na tela inicial, são mostradas as **stars** e **forks** que as pessoas que você segue dão em repositórios públicos, uma funcionalidade que tem por objetivo a ampliação do alcance dos projetos open source.

## Pull Requests (PR's) e Issues

A partir de um **fork**, quando terminar de fazer as alterações que julgar necessárias, você pode submeter uma requisição de anexação do seu conteúdo modificado ao conteúdo original do repositório escolhido. Essa requisição é a **pull request**, que será revisada um maintainer, um dos administradores daquele repositório (no caso de ser um repositório de uma organização, o maintainer pode ser um dos membros dessa organização).

> Também é importante ter em mente que uma contribuição para um projeto open source vai além de modificações diretas no código-fonte. Um feedback, alerta de bugs no software que está sendo desenvolvido, discussões a respeito de um novo passo que pode ser tomado em prol do projeto, troca de ideias e sugestões podem impactar profundamente no futuro do projeto e até da organização. Por isso, é extremamente importante que uma organização fique sempre atenta ao que sua comunidade tem a dizer. Tendo em vista esse contexto, o Github disponibiliza as **issues**, um espaço que pode ser utilizado tanto por usuários comuns quanto por administradores, para a finalidade de todos os pontos citados e também para priorização e organização das tarefas a serem desenvolvidas.

Reviewer 3: Acredito que a continuidade do texto acabou se perdendo sem uma conjunção e poderia ser quebrado em 2 parágrafos.
```suggestion
Também é importante ter em mente que uma contribuição para um projeto open source vai além de modificações diretas no código-fonte, sendo elas, um feedback, alerta de bugs no software que está sendo desenvolvido, discussões a respeito de um novo passo que pode ser tomado em prol do projeto, troca de ideias e sugestões podem impactar profundamente no futuro do projeto e até da organização. Por isso, é extremamente importante que uma organização fique sempre atenta ao que sua comunidade tem a dizer. 
Tendo em vista esse contexto, a plataforma disponibiliza as **issues**, um espaço que pode ser utilizado tanto por usuários comuns quanto por administradores, para a finalidade de todos os pontos citados e também para priorização e organização das tarefas a serem desenvolvidas.
```

> Também é importante ter em mente que uma contribuição para um projeto open source vai além de modificações diretas no código-fonte. Um feedback, alerta de bugs no software que está sendo desenvolvido, discussões a respeito de um novo passo que pode ser tomado em prol do projeto, troca de ideias e sugestões podem impactar profundamente no futuro do projeto e até da organização. Por isso, é extremamente importante que uma organização fique sempre atenta ao que sua comunidade tem a dizer. Tendo em vista esse contexto, o Github disponibiliza as **issues**, um espaço que pode ser utilizado tanto por usuários comuns quanto por administradores, para a finalidade de todos os pontos citados e também para priorização e organização das tarefas a serem desenvolvidas.

Reviewer 4: Acho que era bom desenvolver um pouco sobre as ideias de contribuições, e o que faz uma comunidade: Usuários, Contribuidores e Maintainers.

Ai poderia explicar melhor isso de sugestões e alerta de bugs geralmente são alertados por usuários, que fazem parte da comunidade, e os contribuidores são mais efetivos etc.

> ![Submetendo uma nova issue](https://thepracticaldev.s3.amazonaws.com/i/c7hx2r6s9rhuauhmbsx7.png)

Reviewer 3: Tu acha interessante colocar uma imagem de issue com um texto de descrição de bug ou de melhoria?

Reviewer 4: Acho mais interessante um gif mostrando a submissão. Isso vale pra PR também

## Descobrindo repositórios e contribuindo

> É muito comum que não se saiba muito bem por onde começar a contribuir no início da "aventura" pelo mundo open source. Para minimizar essa dificuldade, você pode visualizar a sessão **discover repositories**, que traz recomendações muito interessantes de repositórios de projetos, baseadas em outros repositórios que você deu star, em pessoas que você segue, em tecnologias recentes que você trabalhou e também em popularidade do projeto, os **trendings**.

Reviewer: > ... você pode visualizar a **seção** discover repositories...

Coloca também um link para essa seção :3

![](https://thepracticaldev.s3.amazonaws.com/i/pjm9maf3lt611kbhjlzr.png)
*Screenshot da página de issues do [Tamburetei](https://github.com/OpenDevUFCG/Tamburetei) com suas labels.*

Você pode contribuir com novas ideias nas discussões a respeito de alguma das issues ou fazer uma PR que a resolva. Posteriormente, quando estiver mais à vontade na organização, você pode abrir novas issues ou tentar resolver algumas mais complexas.

## Explorando o Github

Outra forma de ter acesso a novidades de projetos open source é acessando o **[Explore](https://github.com/explore)**, uma página do Github que além de mostrar projetos recomendados, como mostrado no tópico anterior, também possibilita uma conexão entre as comunidades desses projetos. Você pode explorar conteúdos de curadores, buscar projetos por tópicos e labels, ter acesso a informações de posts, artigos, eventos e meetups relacionados com as tecnologias recomendadas para você, entre outras informações.

![](https://thepracticaldev.s3.amazonaws.com/i/ngl2p0mpmwzd6x8dzziq.png)

> 

Reviewer 3: Senti falta de uma conclusão relacionada ao conteúdo, antes mesmo do agradecimento.

Author: Eu até tinha comentado isso na hora que enviei pra @JRobsonJr e @JuanBarros2 no Discord. Conclusão não é muito o meu forte. Vocês têm alguma sugestão para isso?

Reviewer: Acho que na real poderia concluir convidando a pessoa a ler os textos de @juliobguedes sobre Git (no lugar de botar o disclaimer no começo do texto).

Também é importante ter em mente que uma contribuição para um projeto open source vai além de modificações diretas no código-fonte. Um feedback, alerta de bugs no software que está sendo desenvolvido, discussões a respeito de um novo passo que pode ser tomado em prol do projeto, troca de ideias e sugestões podem impactar profundamente no futuro do projeto e até da organização. Por isso, é extremamente importante que uma organização fique sempre atenta ao que sua comunidade tem a dizer.

Tendo em vista esse contexto, o GitHub disponibiliza as **issues**, um espaço que pode ser utilizado tanto por usuários comuns quanto por administradores, para a finalidade de todos os pontos citados e também para priorização e organização das tarefas a serem desenvolvidas.

![Submetendo uma nova issue](https://thepracticaldev.s3.amazonaws.com/i/c7hx2r6s9rhuauhmbsx7.png)

## Descobrindo repositórios e contribuindo

> É muito comum que não se saiba muito bem por onde começar a contribuir no início da "aventura" pelo mundo open source. Para minimizar essa dificuldade, você pode visualizar a sessão **[discover repositories](https://github.com/discover)**, que traz recomendações muito interessantes de repositórios de projetos, baseadas em outros repositórios que você deu star, em pessoas que você segue, em tecnologias recentes que você trabalhou e também em popularidade do projeto, os **trendings**.

Reviewer 5: "de repositórios de projetos" poderia ficar só "de projeto"

---
title: Contribuindo para projetos open source com GitHub
published: false
description: Uma porta de entrada para pessoas que planejam começar a fazer contribuições para organizações open source.

> tags: opensource, comunidade, github, ptbr.

Reviewer: ```suggestion
tags: opensource, community, github, ptbr.
```

![Submetendo uma nova PR](https://i.imgur.com/DsAq3NW.gif)

Também é importante ter em mente que uma contribuição para um projeto open source vai além de modificações diretas no código-fonte. Um feedback, alerta de bugs no software que está sendo desenvolvido, discussões a respeito de um novo passo que pode ser tomado em prol do projeto, troca de ideias e sugestões podem impactar profundamente no futuro do projeto e até da organização. Por isso, é extremamente importante que uma organização fique sempre atenta ao que sua comunidade tem a dizer.

Tendo em vista esse contexto, o GitHub disponibiliza as **issues**, um espaço que pode ser utilizado tanto por usuários comuns quanto por maintainers, para a finalidade de todos os pontos citados e também para priorização e organização das tarefas a serem desenvolvidas. Observe abaixo o processo para abrir uma nova issue:

![Submetendo uma nova issue](https://i.imgur.com/n0TkrzC.gif)

## Descobrindo repositórios e contribuindo

> É muito comum que não se saiba muito bem por onde começar a contribuir no início da "aventura" pelo mundo open source. Para minimizar essa dificuldade, você pode visualizar a sessão **[discover repositories](https://github.com/discover)**, que traz recomendações muito interessantes de projetos, baseadas em outros repositórios que você deu star, em pessoas que você segue, em tecnologias recentes que você trabalhou e também em popularidade do projeto, os **trendings**.

Reviewer: ```suggestion
É muito comum que não se saiba muito bem por onde começar a contribuir no início da "aventura" pelo mundo open source. Para minimizar essa dificuldade, você pode visualizar a seção **[discover repositories](https://github.com/discover)**, que traz recomendações muito interessantes de projetos, baseadas em outros repositórios que você deu star, em pessoas que você segue, em tecnologias recentes que você trabalhou e também em popularidade do projeto, os **trendings**.
```
